rustls/src/client/tls13.rs

```diff
@@ -904,9 +902,27 @@ impl State<ClientConnectionData> for ExpectFinished {
 
         /* Send our authentication/finished messages.  These are still encrypted
          * with our handshake keys. */
-        if let Some(client_auth) = &mut st.client_auth {
-            emit_certificate_tls13(&mut st.transcript, client_auth, cx.common);
-            emit_certverify_tls13(&mut st.transcript, client_auth, cx.common)?;
+        if let Some(client_auth) = st.client_auth {
+            match client_auth {
+                ClientAuthDetails::Empty {
+                    auth_context_tls13: auth_context,
+                } => {
+                    emit_certificate_tls13(&mut st.transcript, None, auth_context, cx.common);
+                }
+                ClientAuthDetails::Verify {
+                    certkey,
+                    signer,
+                    auth_context_tls13: auth_context,
+                } => {
+                    emit_certificate_tls13(
+                        &mut st.transcript,
+                        Some(&certkey),
+                        auth_context,
+                        cx.common,
+                    );
+                    emit_certverify_tls13(&mut st.transcript, signer.as_ref(), cx.common)?;
+                }
+            }
```

Comment:
I'd probably rewrite this so that there is a single `emit_certificate_tls13()` call.

Reply:
I made the other changes you requested, but I don't think this change would make it clearer, so I didn't make this one.

Reply:
Fair enough, it's definitely less of a win here.